File: cpp/cpp_1000-/1109.cc
// C++
// leetcode 1109
// https://leetcode-cn.com/problems/corporate-flight-bookings/
// 不知道叫啥方法，之前做过类似的思路

class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
    	vector<int>	res(n, 0);
    	for(auto p : bookings) {
    		res[p[0] - 1] += p[2];
    		if(p[1] != n)
    			res[p[1]] -= p[2];
    	}
    	for(int i = 1 ; i < n; i++) {
    		res[i] += res[i - 1];
    	}
    	return	res;
    }
};

File: Go/Go_1000-/1109.go
// Go
// leetcode 1109
// https://leetcode.cn/problems/corporate-flight-bookings/

func corpFlightBookings(bookings [][]int, n int) []int {
	res := make([]int, n)

	for _, order := range bookings {
		res[order[0]-1] += order[2]
		if order[1] < n {
			res[order[1]] -= order[2]
		}
	}

	for i := 1; i < n; i++ {
		res[i] += res[i-1]
	}
	return res
}